<?xml version="1.0"?>
<sdf version="1.6">
  <model name="cylinders">
    <static>false</static>
    
    <!-- First Cylinder (Pink) -->
    <link name="cylinder1">
      <pose>0 0 0.5 0 0 0</pose>  <!-- Initial position -->
      <visual name="visual1">
        <geometry><cylinder><radius>0.35</radius><length>1.6</length></cylinder></geometry>
        <material>
          <ambient>0.7 0.0 0.35 1.0</ambient>  <!-- Darker pink -->
          <diffuse>0.7 0.57 0.64 1.0</diffuse>   <!-- Less bright -->
          <specular>0.1 0.1 0.1 1.0</specular>   <!-- Low reflectivity -->
          <emissive>0.0 0.0 0.0 1.0</emissive>
        </material>
      </visual>
      <collision name="collision1">
        <geometry>
          <cylinder><radius>0.35</radius><length>1.6</length></cylinder>
        </geometry>
        <surface>
          <contact>
            <ode>
              <min_depth>0.01</min_depth>  <!-- Ensures LiDAR detects small objects -->
            </ode>
          </contact>
        </surface>
    </collision>
    </link>

    <!-- Second Cylinder (Blue) -->
    <link name="cylinder2">
      <pose>0 0 2.0 0 0 0</pose>  <!-- Initial position -->
      <visual name="visual2">
        <geometry><cylinder><radius>0.35</radius><length>1.6</length></cylinder></geometry>
        <material>
          <ambient>0.0 0.35 0.7 1.0</ambient>    <!-- Darker blue -->
          <diffuse>0.57 0.64 0.7 1.0</diffuse>    <!-- Less bright -->
          <specular>0.1 0.1 0.1 1.0</specular>    <!-- Low reflectivity -->
          <emissive>0.0 0.0 0.0 1.0</emissive>
        </material>
      </visual>
      <collision name="collision2">
        <geometry>
          <cylinder><radius>0.35</radius><length>1.6</length></cylinder>
        </geometry>
        <surface>
          <contact>
            <ode>
              <min_depth>0.01</min_depth>  <!-- Ensures LiDAR detects small objects -->
            </ode>
          </contact>
        </surface>
    </collision>
    </link>

    <!-- third Cylinder (green) -->
    <link name="cylinder3">
      <pose>0 0 2.0 0 0 0</pose>  <!-- Initial position -->
      <visual name="visual3">
        <geometry><cylinder><radius>0.35</radius><length>1.6</length></cylinder></geometry>
        <material>
          <ambient>0.0 0.35 0.0 1.0</ambient>    <!-- Darker green -->
          <diffuse>0.2 0.7 0.2 1.0</diffuse>      <!-- Medium green -->
          <specular>0.1 0.1 0.1 1.0</specular>    <!-- Low reflectivity -->
          <emissive>0.0 0.0 0.0 1.0</emissive>
        </material>
      </visual>
      <collision name="collision3">
        <geometry>
          <cylinder><radius>0.35</radius><length>1.6</length></cylinder>
        </geometry>
        <surface>
          <contact>
            <ode>
              <min_depth>0.01</min_depth>  <!-- Ensures LiDAR detects small objects -->
            </ode>
          </contact>
        </surface>
    </collision>
    </link>
  </model>
</sdf>